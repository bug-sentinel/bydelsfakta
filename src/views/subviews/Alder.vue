<template>
  <div class="main-container">
    <div class="main-container__cards">
      <graph-card v-for="(card, index) in settings.median" :key="`alder-median-${index}`" :settings="card" />
      <graph-card v-for="(card, index) in settings.populasjonspyramide" :key="`alder-pyramide-${index}`"  :settings="card" />
      <graph-card v-for="(card, index) in settings.aldersgrupper" :key="`alder-grupper-${index}`"  :settings="card" />
    </div>
    <div class="main-container__map">
      <h3>Kart</h3>
    </div>
  </div>
</template>

<script>
import GraphCard from '../../components/GraphCard.vue';

export default {
  name: 'Alder',
  components: { GraphCard },
  props: {
    settings: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style scoped>
</style>
