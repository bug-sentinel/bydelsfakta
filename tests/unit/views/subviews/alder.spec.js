import { mount, createLocalVue } from '@vue/test-utils';
import VueResize from 'vue-resize';
import Alder from '../../../../src/views/subviews/Alder.vue';
import page from '../../../../src/config/pages';
import clickOutside from '../../../../src/directives/clickOutside';

describe('Alder', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(VueResize);
    localVue.directive('click-outside', clickOutside);
    wrapper = mount(Alder, {
      propsData: {
        settings: page['alder'],
      },
      localVue,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders Alder component correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
