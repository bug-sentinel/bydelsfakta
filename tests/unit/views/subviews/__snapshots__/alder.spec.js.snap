// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alder renders Alder component correctly 1`] = `
<div
  class="main-container"
>
  <div
    class="main-container__cards"
  >
    <h2
      class="section-heading"
    >
      Utvalgte aldersgrupper
    </h2>
     
    <div
      class="card-container large"
    >
      <div
        class="card"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Antall"
            aria-selected="[object Object]"
            class="tabs__button active"
            id="tabButton-0"
            role="tab"
          >
            Antall
          </button>
          <button
            aria-label="Andel"
            aria-selected="[object Object]"
            class="tabs__button"
            id="tabButton-1"
            role="tab"
          >
            Andel
          </button>
           
          <div
            class="context-menu"
            role="menu"
          >
            <button
              aria-haspopup="true"
              aria-label="åpne meny"
              class="context-menu__button"
              id="context-menu-button"
            >
              <i
                class="material-icons"
              >
                menu
              </i>
            </button>
             
            <!---->
          </div>
        </div>
         
        <div
          class="graph__container"
        >
          <svg
            class="graph__svg"
            style="font-family: OsloSans;"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <text
              class="heading"
              font-size="14"
              font-weight="bold"
              y="14"
            />
            <g
              class="canvas"
              transform="translate(0, 50)"
            >
              <g
                class="axis x"
              />
              <g
                class="axis y"
              />
              <g
                class="axis x2"
              />
              <g
                class="axis y2"
              />
              <g
                class="upper"
              >
                <g
                  class="brush"
                  fill="none"
                  pointer-events="all"
                  transform="translate(160, 0)"
                >
                  <rect
                    class="overlay"
                    cursor="crosshair"
                    height="100"
                    pointer-events="all"
                    width="840"
                    x="0"
                    y="0"
                  />
                  <rect
                    class="selection"
                    cursor="move"
                    fill="#F8C66B"
                    fill-opacity="0.15"
                    shape-rendering="crispEdges"
                    stroke="#F8C66B"
                    stroke-opacity="1"
                    stroke-width="2"
                    style="display: none;"
                  />
                  <rect
                    class="handle handle--e"
                    cursor="ew-resize"
                    style="display: none;"
                  />
                  <rect
                    class="handle handle--w"
                    cursor="ew-resize"
                    style="display: none;"
                  />
                </g>
                <g
                  class="axis x"
                  transform="translate(160, 100)"
                />
                <g
                  class="axis y"
                  transform="translate(160, 0)"
                />
              </g>
              <g
                class="middle"
                transform="translate(0, 140)"
              >
                <g
                  class="brush"
                  fill="none"
                  pointer-events="all"
                  transform="translate(160, 0)"
                >
                  <rect
                    class="overlay"
                    cursor="crosshair"
                    height="19"
                    pointer-events="all"
                    rx="6"
                    stroke="#292858"
                    width="840"
                    x="0"
                    y="0"
                  />
                  <rect
                    class="selection"
                    cursor="move"
                    fill="#F8C66B"
                    fill-opacity="1"
                    shape-rendering="crispEdges"
                    stroke="#fff"
                    style="display: none;"
                  />
                  <rect
                    class="handle handle--e"
                    cursor="ew-resize"
                    style="display: none;"
                  />
                  <rect
                    class="handle handle--w"
                    cursor="ew-resize"
                    style="display: none;"
                  />
                  <g
                    class="handleIcon"
                    transform="translate(0, -9)"
                  >
                    <path
                      d="M21 0H10C4 0 0 4 0 10v17c0 6 4 10 10 10h11V0z"
                      fill="#292858"
                      style="pointer-events: none;"
                    />
                    <rect
                      fill="white"
                      fill-opacity="0.75"
                      height="11"
                      style="pointer-events: none;"
                      width="1"
                      x="6"
                      y="13"
                    />
                    <rect
                      fill="white"
                      fill-opacity="0.75"
                      height="11"
                      style="pointer-events: none;"
                      width="1"
                      x="10"
                      y="13"
                    />
                    <rect
                      fill="white"
                      fill-opacity="0.75"
                      height="11"
                      style="pointer-events: none;"
                      width="1"
                      x="14"
                      y="13"
                    />
                  </g>
                  <g
                    class="handleIcon"
                    transform="translate(0, -9)"
                  >
                    <path
                      d="M0 0h11c6 0 10 4 10 10v17c0 6-4 10-10 10H0V0z"
                      fill="#292858"
                      style="pointer-events: none;"
                    />
                    <rect
                      fill="white"
                      fill-opacity="0.75"
                      height="11"
                      style="pointer-events: none;"
                      width="1"
                      x="6"
                      y="13"
                    />
                    <rect
                      fill="white"
                      fill-opacity="0.75"
                      height="11"
                      style="pointer-events: none;"
                      width="1"
                      x="10"
                      y="13"
                    />
                    <rect
                      fill="white"
                      fill-opacity="0.75"
                      height="11"
                      style="pointer-events: none;"
                      width="1"
                      x="14"
                      y="13"
                    />
                  </g>
                </g>
              </g>
              <g
                class="lower"
                transform="translate(0, 230)"
              >
                <text
                  class="xAxis-title"
                  fill="#292858"
                  font-size="12"
                  font-weight="700"
                  transform="translate(300, -28)"
                />
                <g
                  class="axis x"
                  transform="translate(300, 0)"
                />
              </g>
            </g>
            <g
              class="sourceGroup"
              fill="#292858"
              opacity="0"
            >
              <text
                class="source-label"
                font-size="10"
              >
                Kilde: 
              </text>
              <text
                class="source"
                font-size="10"
              />
            </g>
          </svg>
           
          <div
            class="resize-observer"
            data-v-b329ee4c=""
            tabindex="-1"
          >
            <object
              aria-hidden="true"
              data="about:blank"
              tabindex="-1"
              type="text/html"
            />
          </div>
          <div
            class="graph__dropdown"
          >
            <label
              class="graph__dropdown__label"
              for="age_selector"
            >
              Velg segment
            </label>
            <select
              class="graph__dropdown__select"
              id="age_selector"
            >
              <option
                value="[0, 119]"
              >
                Alle
              </option>
              <option
                value="[0, 1]"
              >
                0–1 år
              </option>
              <option
                value="[0, 5]"
              >
                0–5 år
              </option>
              <option
                value="[6, 12]"
              >
                6–12 år
              </option>
              <option
                value="[6, 15]"
              >
                6–15 år
              </option>
              <option
                value="[6, 17]"
              >
                6–17 år
              </option>
              <option
                value="[12, 15]"
              >
                12–15 år
              </option>
              <option
                value="[16, 18]"
              >
                16–18 år
              </option>
              <option
                value="[16, 18]"
              >
                19–29 år
              </option>
              <option
                value="[16, 18]"
              >
                19–29 år
              </option>
              <option
                value="[30, 49]"
              >
                30–49 år
              </option>
              <option
                value="[50, 66]"
              >
                50–66 år
              </option>
              <option
                value="[67, 119]"
              >
                67+ år
              </option>
              <option
                value="[60, 79]"
              >
                60–79 år
              </option>
              <option
                value="[80, 119]"
              >
                80+ år
              </option>
              <option
                value="[80, 89]"
              >
                80–89 år
              </option>
              <option
                value="[90, 119]"
              >
                90+ år
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
     
    <h2
      class="section-heading"
    >
      Medianalder i Bydel BYDELSNAVN
    </h2>
     
    <div
      class="card-container large"
    >
      <div
        class="card"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Status"
            aria-selected="[object Object]"
            class="tabs__button active"
            id="tabButton-0"
            role="tab"
          >
            Status
          </button>
           
          <div
            class="context-menu"
            role="menu"
          >
            <button
              aria-haspopup="true"
              aria-label="åpne meny"
              class="context-menu__button"
              id="context-menu-button"
            >
              <i
                class="material-icons"
              >
                menu
              </i>
            </button>
             
            <!---->
          </div>
        </div>
         
        <div
          class="graph__container"
        >
          <svg
            class="graph__svg"
            style="font-family: OsloSans;"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <text
              class="heading"
              font-size="14"
              font-weight="bold"
              y="14"
            />
            <g
              class="canvas"
              transform="translate(240, 90)"
            >
              <g
                class="axis x"
              />
              <g
                class="axis y"
              />
              <g
                class="axis x2"
              />
              <g
                class="axis y2"
              />
              <text
                class="label-median"
                font-size="13"
                text-anchor="middle"
                x="-175"
                y="-40"
              >
                Aldersfordeling og medianalder
              </text>
              <text
                class="label-mean"
                font-size="13"
                text-anchor="middle"
                x="-85"
                y="-40"
              >
                Gjennomsnittsalder
              </text>
            </g>
            <g
              class="sourceGroup"
              fill="#292858"
              opacity="0"
            >
              <text
                class="source-label"
                font-size="10"
              >
                Kilde: 
              </text>
              <text
                class="source"
                font-size="10"
              />
            </g>
          </svg>
           
          <div
            class="resize-observer"
            data-v-b329ee4c=""
            tabindex="-1"
          >
            <object
              aria-hidden="true"
              data="about:blank"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
      </div>
    </div>
     
    <h2
      class="section-heading"
    >
      Populasjonspyramide
    </h2>
     
    <div
      class="card-container large"
    >
      <div
        class="card"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Status"
            aria-selected="[object Object]"
            class="tabs__button active"
            id="tabButton-0"
            role="tab"
          >
            Status
          </button>
          <button
            aria-label="Historisk"
            aria-selected="[object Object]"
            class="tabs__button"
            id="tabButton-1"
            role="tab"
          >
            Historisk
          </button>
           
          <div
            class="context-menu"
            role="menu"
          >
            <button
              aria-haspopup="true"
              aria-label="åpne meny"
              class="context-menu__button"
              id="context-menu-button"
            >
              <i
                class="material-icons"
              >
                menu
              </i>
            </button>
             
            <!---->
          </div>
        </div>
         
        <div
          class="graph__container"
        >
          <svg
            class="graph__svg"
            style="font-family: OsloSans;"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <text
              class="heading"
              font-size="14"
              font-weight="bold"
              y="14"
            />
            <g
              class="canvas"
              transform="translate(300, 70)"
            >
              <g
                class="pyramid"
              />
              <g
                class="axis y axis-left"
              />
              <g
                class="axis y axis-right"
              />
              <g
                class="axis y axis-lines"
              />
              <g
                class="axis x"
                transform="translate(0, 0)"
              />
              <g
                class="yAxis-title"
                transform="translate(-50, 0)"
              >
                <text
                  fill="#292858"
                  font-size="12"
                  font-weight="700"
                  text-anchor="middle"
                  transform="rotate(-90)"
                >
                  Alder
                </text>
              </g>
              <text
                class="xAxis-title"
                fill="#292858"
                font-size="12"
                font-weight="700"
                text-anchor="middle"
                transform="translate(0, 36)"
              >
                Folkemengde
              </text>
            </g>
            <g
              class="list"
              transform="translate(3, 70)"
            >
              <text
                fill="#292858"
                font-size="12"
                font-weight="700"
                style="text-transform: uppercase;"
                transform="translate(10, -16)"
              >
                Velg delbydel
              </text>
            </g>
            <g
              class="sourceGroup"
              fill="#292858"
              opacity="0"
            >
              <text
                class="source-label"
                font-size="10"
              >
                Kilde: 
              </text>
              <text
                class="source"
                font-size="10"
              />
            </g>
          </svg>
           
          <div
            class="resize-observer"
            data-v-b329ee4c=""
            tabindex="-1"
          >
            <object
              aria-hidden="true"
              data="about:blank"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="main-container__map"
  >
    <h3>
      Kart
    </h3>
  </div>
</div>
`;
